internal/provider/kubernetes/controller.go

```diff
@@ -1465,6 +1466,60 @@ func (r *gatewayAPIReconciler) processParamsRef(ctx context.Context, gc *gwapiv1
 				validationErr = fmt.Errorf("invalid envoyproxy: %w", err)
 				continue
 			}
+
```

Comment:
we probably also need to enhance https://github.com/envoyproxy/gateway/blob/e5f86303d83d732f8c3890a6868c62e29a3e3029/internal/provider/kubernetes/predicates.go#L217 to make sure we trigger a reconcile when the `Service` changes

Reply:
yeah, but want to do that in an seperated PR.